source/lex.tex

```diff
@@ -1007,28 +1009,29 @@
 \indextext{literal!\idxcode{unsigned}}%
 \indextext{literal!\idxcode{long}}%
 \indextext{literal!base of integer}%
-An \defnx{integer literal}{literal!integer} is a sequence of digits that has no period
-or exponent part, with optional separating single quotes that are ignored
-when determining its value. An integer literal may have a prefix that specifies
-its base and a suffix that specifies its type. The lexically first digit
-of the sequence of digits is the most significant.
-A \defnx{binary integer literal}{literal!binary} (base two) begins with
-\tcode{0b} or \tcode{0B} and consists of a sequence of binary digits.
-An \defnx{octal integer literal}{literal!octal}
-(base eight) begins with the digit \tcode{0} and consists of a
-sequence of octal digits.\footnote{The digits \tcode{8} and \tcode{9} are not octal digits. }
-A \defnx{decimal integer literal}{literal!decimal}
-(base ten) begins with a digit other than \tcode{0} and
-consists of a sequence of decimal digits.
-A \defnx{hexadecimal integer literal}{literal!hexadecimal}
-(base sixteen) begins with
-\tcode{0x} or \tcode{0X} and consists of a sequence of hexadecimal
-digits, which include the decimal digits and the letters \tcode{a}
-through \tcode{f} and \tcode{A} through \tcode{F} with decimal values
-ten through fifteen.
+In an \grammarterm{integer-literal},
+the sequence of digits is interpreted as a base-$N$ number
```

Comment:
We've lost the connection between the word "digits" and the grammar. Perhaps

> the sequence of *binary-digit*s, *octal-digit*s, *digit*s, or *hexadecimal-digit*s is interpreted as a base-$N$ number
> as shown in table [tab:lex.icon.base];
> the lexically first digit of the sequence of digits is the most significant.
> [Note: The prefix and any optional separating single quotes are ignored when determining the value.]

Reply:
Sounds good.